keps/sig-storage/2268-non-graceful-shutdown/README.md

```diff
@@ -286,15 +323,26 @@ _This section must be completed when targeting beta graduation to a release._
   Ideally, this should be a metric. Operations against the Kubernetes API (e.g.,
   checking if there are objects with field X set) may be a last resort. Avoid
   logs or events for this purpose.
+  An operator, the person who operates the cluster, can check if the
+  `NodeOutOfServiceVolumeDetach` feature gate is enabled and if there is an
+  `out-of-service` taint on the shutdown node.
+  The usage of this feature requires the manual step of applying a taint
+  so the operator should be the one applying it.
 
 * **What are the SLIs (Service Level Indicators) an operator can use to determine
 the health of the service?**
   - [ ] Metrics
-    - Metric name:
+    - Metric name: We can add new metrics deleting_pods_total, deleting_pods_error_total
+      in Pod GC Controller.
+      For Attach Detach Controller, there's already a metric:
+      attachdetach_controller_forced_detaches. We could keep it for force detach
+      by timeout and add a new metric attachdetach_controller_forced_detaches_taint.
```

Comment:
I perhaps wouldn't bother implementing new kube metrics for this one. What an admin really needs to watch for is, number of `NotReady` nodes in a cluster and add `out-of-service` taints, if they are not going to come back immediately and similarly monitor nodes with `out-of-service` taint and remove the taint if node is up and running but still has the taint.

Reply:
Add a new metric to indicate the number of nodes with taints?  Maybe kube-state-metrics is the project to add this?

Reply:
I suggested it could be something we can explore with [kube-state-metrics](https://github.com/kubernetes/kube-state-metrics). It produces metrics based on k8s object states.

Reply:
Ok, I'll add this to the KEP.